Kernel/Syscalls/posix_spawn.cpp

```diff
@@ -0,0 +1,152 @@
+    TRY(child->m_fds.with_exclusive([&](auto& fds) -> ErrorOr<void> {
+        TRY(fds.try_resize(Process::OpenFileDescriptions::max_open()));
+
+        // NOTE: If Device::base_devices() is returning nullptr, it means the null device is not attached which is a bug.
+        VERIFY(Device::base_devices() != nullptr);
+        auto& device_to_use_as_tty = tty() ? (CharacterDevice&)*tty() : Device::base_devices()->null_device;
+        auto description = TRY(device_to_use_as_tty.open(O_RDWR));
+        auto setup_description = [&](int fd) {
+            fds.m_fds_metadatas[fd].allocate();
+            fds[fd].set(*description);
+        };
+        setup_description(0);
+        setup_description(1);
+        setup_description(2);
+
+        return {};
+    }));
+
+    // Copy protected data
+    // TODO: we are probably over-copying here
+    with_protected_data([&](auto& my_protected_data) {
+        child->with_mutable_protected_data([&](auto& child_protected_data) {
+            child_protected_data.promises = my_protected_data.promises;
+            child_protected_data.execpromises = my_protected_data.execpromises;
+            child_protected_data.has_promises = my_protected_data.has_promises;
+            child_protected_data.has_execpromises = my_protected_data.has_execpromises;
+            child_protected_data.credentials = my_protected_data.credentials;
+            child_protected_data.umask = my_protected_data.umask;
+            child_protected_data.dumpable = my_protected_data.dumpable;
+            child_protected_data.process_group = my_protected_data.process_group;
+            // NOTE: Propagate jailed_until_exit property to child processes.
+            // The jailed_until_exec property is also propagated, but will be
+            // set to false once the child process is calling the execve syscall.
+            if (my_protected_data.jailed_until_exit.was_set())
+                child_protected_data.jailed_until_exit.set();
+            child_protected_data.jailed_until_exec = my_protected_data.jailed_until_exec;
+        });
+    });
+
+    dbgln_if(FORK_DEBUG, "posix_spawn: child={}", child);
+
+    // A child created via fork(2) inherits a copy of its parent's signal mask
+    child_first_thread->update_signal_mask(Thread::current()->signal_mask());
+
+    // A child process created via fork(2) inherits a copy of its parent's alternate signal stack settings.
+    child_first_thread->m_alternative_signal_stack = Thread::current()->m_alternative_signal_stack;
+
+    thread_finalizer_guard.disarm(); // TODO: this probably isnt in the right place?
+
+    m_scoped_process_list.with([&](auto& list_ptr) {
+        if (list_ptr) {
+            child->m_scoped_process_list.with([&](auto& child_list_ptr) {
+                child_list_ptr = list_ptr;
+            });
+            list_ptr->attach(*child);
+        }
+    });
+
+    Thread* new_main_thread = nullptr;
```

Comment:
🤔 didn't we already create a main thread for the process? If we don't intend to re-use it in this patch, a FIXME would be appropriate here.